### PR TITLE
Add utility and makefile type in gmake.

### PR DESCRIPTION
**Pull request by tvandijck**

### CODE REVIEW

File: src/actions/make/make_makefile.lua [hunk 1]
```diff
@@ -0,0 +1,93 @@
+--
+-- make_makefile.lua
+-- Generate a C/C++ project makefile.
```
Comment (tritao) [inline src/actions/make/make_makefile.lua]: This needs to be updated.

File: src/actions/make/make_utility.lua [hunk 2]
```diff
@@ -0,0 +1,67 @@
+--
+-- make_utility.lua
+-- Generate a C/C++ project makefile.
```
Comment (tritao) [inline src/actions/make/make_utility.lua]: Same.

File: src/actions/make/make_cpp.lua [hunk 3]
```diff
@@ -98,6 +99,14 @@
 	end
 
 
+	function make.exePaths(cfg)
```
Comment (tritao) [inline src/actions/make/make_cpp.lua]: Is this related to the Utility and Makefile kinds changes?

**Comment by starkos** [comment]
These would be great to have. The only thing that jumps out at me is we have separate calls for things like `makefileTargetRules()` and `utilityTargetRules()` while in the VS actions that would just be `targetRules()` with a if-else statement for the different project types. Not sure if one approach is really better than another of it they are just different.

File: src/actions/make/make_makefile.lua (hunk 1 above)
Comment (tvandijck) [inline src/actions/make/make_makefile.lua]: updated to what?

File: src/actions/make/make_cpp.lua (hunk 3 above)
Comment (tvandijck) [inline src/actions/make/make_cpp.lua]: unrelated, but I kind of had a hard time extracting it... it basically just adds support for "binpath", just like in MSVC.

**Comment by tvandijck** [comment]
I have no opinion on what is better, I don't understand enough of the gmake backend to make a ruling on that.... we just found it easier to separate and be explicit...

File: src/actions/make/make_makefile.lua (hunk 1 above)
Comment (tritao) [inline src/actions/make/make_makefile.lua]: At the time it seemed to me like a copy pasted description, the Utility and Makefile types should not be C/C++ project related right?

**Comment by Trass3r** [comment]
What is the status of this?

**Comment by tvandijck** [comment]
It's working for our use-cases.... it may not for many other use-cases... but it really just needs to be merged or declined... as I said, I have no opinion so I'm not clicking on the "merge" button.

**Comment by Trass3r** [comment]
Well I guess having it in is way better than not as long as it exits if any assertions you made are violated.

**Comment by starkos** [comment]
I haven't merged because I want to think about the API issues I mentioned in my earlier comment, and I was hoping someone else might chime in. I will go ahead and merge it.

**Comment by tvandijck** [comment]
a cleanup/refactor pass is definitely in order to make this more similar to the visual studio action.. I'm swamped in work though, so I just don't have the time for this right now. anyone else is obviously welcome to contribute...

**Pull request merged**